.github/workflows/run-pytest.yml

```diff
@@ -5,38 +5,54 @@ permissions:
   contents: read # to fetch code (actions/checkout)
 
 jobs:
-  build-all:
+#  build-all:
+#    runs-on: ubuntu-latest
+#    strategy:
+#      fail-fast: true
+#      matrix:
+#        python-version: ['3.8', '3.9', '3.10', '3.11', '3.12']
+#        numpy-version: ['>=1.21,<2.0', '>=2.0']
+#        exclude:
+#          - python-version: '3.8'  # numpy>=2.0 requires Python>=3.9
+#            numpy-version: '>=2.0'
+#    steps:
+#      - uses: actions/checkout@v4
+#      - run: |
+#          docker build -f bin/all-py.Dockerfile \
+#            --build-arg PYTHON_VERSION="${{ matrix.python-version }}" \
+#            --build-arg NUMPY_VERSION="${{ matrix.numpy-version }}" \
+#            --tag gymnasium-all-docker .
+#      - name: Run tests
+#        run: docker run gymnasium-all-docker pytest tests/*
+#      - name: Run doctests
+#        if: ${{ matrix.python-version != '3.8' }}
+#        run: docker run gymnasium-all-docker pytest --doctest-modules gymnasium/
+#
+#  build-necessary:
+#    runs-on:
+#      ubuntu-latest
+#    steps:
+#      - uses: actions/checkout@v4
+#      - run: |
+#           docker build -f bin/necessary-py.Dockerfile \
+#             --build-arg PYTHON_VERSION='3.10' \
+#             --tag gymnasium-necessary-docker .
+#      - name: Run tests
+#        run: |
+#           docker run gymnasium-necessary-docker pytest tests/test_core.py tests/envs/test_envs.py tests/spaces
+#
+  build-type-testing:
     runs-on: ubuntu-latest
     strategy:
       fail-fast: true
       matrix:
-        python-version: ['3.8', '3.9', '3.10', '3.11', '3.12']
         numpy-version: ['>=1.21,<2.0', '>=2.0']
-        exclude:
-          - python-version: '3.8'  # numpy>=2.0 requires Python>=3.9
-            numpy-version: '>=2.0'
     steps:
       - uses: actions/checkout@v4
       - run: |
           docker build -f bin/all-py.Dockerfile \
-            --build-arg PYTHON_VERSION="${{ matrix.python-version }}" \
+            --build-arg PYTHON_VERSION="3.12" \
```

Comment:
You are fixing some issues on python 3.12, But those changes might be breaking for Python 3.8, 

Typings should be tested on the lowest supported version.